src/psfmachine/ffi.py

```diff
@@ -0,0 +1,924 @@
+"""Subclass of `Machine` that Specifically work with FFIs"""
+import os
+import numpy as np
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+
+from astropy.io import fits
+from astropy.stats import SigmaClip
+from astropy.time import Time
+from astropy.wcs import WCS
+from photutils import Background2D, MedianBackground, BkgZoomInterpolator
+
+# from . import PACKAGEDIR
+from .utils import do_tiled_query
+from .machine import Machine
+from .version import __version__
+
+__all__ = ["FFIMachine"]
+
+
+class FFIMachine(Machine):
+    """
+    Subclass of Machine for working with FFI data. It is a subclass of Machine
+    """
+
+    def __init__(
+        self,
+        channel=1,
+        quarter=5,
+        wcs=None,
+        limit_radius=32.0,
+        n_r_knots=10,
+        n_phi_knots=15,
+        cut_r=6,
+        rmin=1,
+        rmax=16,
+        **kwargs,
+    ):
+        """
+        Class to work with FFI data.
+
+        Parameters
+        ----------
+        channel : int
+            Channel number to be used
+        quarter : int
+            Quarter/Campagn nunmber to be used (for Kepler data).
+        wcs : astropy.wcs
+            World coordinates system solution for the FFI. Used for plotting.
+        **kwargs
+            Keyword attributes that contain information parsed from `from_file()` and
+            is used to initialize a `Machine` class object.
+
+        Attributes
+        ----------
+        All attributes inherited from Machine.
+
+        meta : dictionary
+            Meta data information related to the FFI
+        wcs : astropy.wcs
+            World coordinates system solution for the FFI. Used for plotting.
+        flux_2d : numpy.ndarray
+            2D image representation of the FFI, used for plotting.
+        channel : int
+            Channel number to be used
+        quarter : int
+            Quarter/Campagn nunmber to be used (for Kepler data).
+        """
+        self.column = kwargs["column"].ravel()
+        self.row = kwargs["row"].ravel()
+        self.ra = kwargs["ra"].ravel()
+        self.dec = kwargs["dec"].ravel()
+        # keep 2d image for easy plotting
+        self.flux_2d = kwargs["flux"]
+        # reshape flux and flux_err as [ntimes, npix]
+        self.flux = kwargs["flux"].reshape(kwargs["flux"].shape[0], -1)
```

Comment:
It seems like you're putting the flux into kwargs. This is harder for the user, because 1) there's no prompt for flux, and if I don't pass flux this will break. 2) there is no docstring for flux. Let's be explicit. Make sure all the keyword arguments that are required in the init are required arguments for `__init__`

Reply:
You're spending memory here to have a second `flux` attribute. Why not have flux be 2D and then reshape it whenever necessary? or have flux be 1D, and reshape when necessary?

Reply:
Could add static method for this e.g.
```python
@staticmethod
def flux_unwrap(self):
     return self.flux[apply_mask]
```
depends on what you want